discovery-provider/src/tasks/index_network_peers.py

```diff
@@ -0,0 +1,174 @@
+import logging
+import concurrent.futures
+from urllib.parse import urljoin
+import requests
+from src.tasks.celery_app import celery
+from src import eth_abi_values
+from src.utils.helpers import get_ipfs_info_from_cnode_endpoint
+from src.models import User
+from src.utils.redis_cache import use_redis_cache
+
+logger = logging.getLogger(__name__)
+
+sp_factory_registry_key = bytes("ServiceProviderFactory", "utf-8")
+content_node_service_type = bytes("creator-node", "utf-8")
+
+# What is a "Peer" in this context?
+# A peer represents another known entity in the network
+# The logic here is to ensure a robust connection from an active indexer
+# to all active entities in the network.
+# This is to ensure minimal retrieval time within the actual indexing flow itself
+# NOTE - The terminology of "peer" in this file overlaps with ipfs swarm peers
+#   Even though we 'swarm connect' to an ipfs node embedded within our protocol the
+#   concept is very much distinct.
+
+# Perform eth web3 call to fetch endpoint info
+def fetch_cnode_info(sp_id, sp_factory_instance):
+    cn_endpoint_info = sp_factory_instance.functions.getServiceEndpointInfo(
+        content_node_service_type,
+        sp_id
+    ).call()
+    return cn_endpoint_info
+
+# Query the L1 set of audius protocol contracts and retrieve a list of peer endpoints
+def retrieve_peers_from_eth_contracts(self):
+    shared_config = update_network_peers.shared_config
+    eth_web3 = update_network_peers.eth_web3
+    eth_registry_address = update_network_peers.eth_web3.toChecksumAddress(
+        shared_config["eth_contracts"]["registry"]
+    )
+    eth_registry_instance = eth_web3.eth.contract(
+        address=eth_registry_address, abi=eth_abi_values["Registry"]["abi"]
+    )
+    sp_factory_address = eth_registry_instance.functions.getContract(
+        sp_factory_registry_key
+    ).call()
+    sp_factory_inst = eth_web3.eth.contract(
+        address=sp_factory_address, abi=eth_abi_values["ServiceProviderFactory"]["abi"]
+    )
+    total_cn_type_providers = sp_factory_inst.functions.getTotalServiceTypeProviders(content_node_service_type).call()
+    ids_list = list(range(1, total_cn_type_providers + 1))
```

Comment:
its implicit here that you are generating the spIDs with range of the list totalProv length
isn't this wrong? what if a node de-registers? to be safe we should actually fetch the ids with a map over `total_cn_type_providers`

Reply:
i think i see what you're saying and if you mean filter out deregistered endpoints so we don't intentionally fail peering connections - i totally agree. but let's confirm this tmrw

Reply:
resolution to this - i have added a legit FQDN check